Sources/swift-format/Utilities/DiagnosticsEngine.swift

```diff
@@ -65,6 +65,20 @@ final class DiagnosticsEngine {
         message: message))
   }
 
+  /// Emits a generic warning message.
+  ///
+  /// - Parameters:
+  ///   - message: The message associated with the error.
+  ///   - location: The `Diagnostic.Location` in the source code associated with the error, or nil if there is no
+  ///     location associated with the error.
+  func emitWarning(_ message: String, location: Diagnostic.Location? = nil) {
```

Comment:
I'm not _entirely_ happy with this — this function is very similar to `emitError`, but it takes `Diagnostic.Location?` instead of `SourceLocation`. Nothing wrong with that, but the API now is slightly inconsistent. 

I don't know if it's worth providing the other two overloads for consistency, or if that's fine.